lib/galaxy/jobs/runners/kubernetes.py

```diff
@@ -772,7 +772,10 @@ def check_watched_item(self, job_state):
             # as probably this means that the k8s API server hasn't
             # had time to fill in the object status since the
             # job was created only too recently.
-            if len(job.obj["status"]) == 0:
+            # It is possible that k8s didn't account for the status of the pods
+            # and they are in the uncountedTerminatedPods status. In this
+            # case we also need to wait a moment
+            if len(job.obj["status"]) == 0 or "uncountedTerminatedPods" in job.obj["status"]:
```

Comment:
The kubernetes tests are timing out. Should the check maybe be
```suggestion
            if len(job.obj["status"]) == 0 or in job.obj["status"].get("uncountedTerminatedPods"):
```
{"status": {"uncountedTerminatedPods": []} or something like this. In fact is there an API spec for te status object ?

Reply:
https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.30/#jobstatus-v1-batch
This is the best reference I found, and it marked as optional here: https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/V1JobStatus.md. 

For me it was either there and non-empty; or not there at all. I cannot be certain how it looks like across various versions etc.; yours looks more stable in this regard. I'll commit your suggestion :)